ClearKeep/App/System/AppDelegate.swift

```diff
@@ -55,4 +85,107 @@ final class AppDelegate: UIResponder, UIApplicationDelegate {
 			.compactMap({ $0.windowScene?.delegate as? SceneDelegate })
 			.first
 	}
+	
+	func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey: Any] = [:]) -> Bool {
+		// Facebook
+		ApplicationDelegate.shared.application(app, open: url, sourceApplication: options[UIApplication.OpenURLOptionsKey.sourceApplication] as? String, annotation: options[UIApplication.OpenURLOptionsKey.annotation])
+		
+		// Office365
+		MSALPublicClientApplication.handleMSALResponse(url, sourceApplication: options[UIApplication.OpenURLOptionsKey.sourceApplication] as? String)
+		
+		// Google
+		let handled = GIDSignIn.sharedInstance.handle(url)
+		if handled {
+			return true
+		}
+		
+		// Handle other custom URL types.
+		
+		// If not handled by this app, return false.
+		return false
+	}
+}
+
+extension AppDelegate: PKPushRegistryDelegate {
+	fileprivate func registrationPushRegistry() {
+		let voipRegistry: PKPushRegistry = PKPushRegistry(queue: .main)
+		
+		voipRegistry.delegate = self
+		
+		voipRegistry.desiredPushTypes = [PKPushType.voIP]
+	}
+	
+	func pushRegistry(_ registry: PKPushRegistry, didUpdate pushCredentials: PKPushCredentials, for type: PKPushType) {
+		systemEventsHandler?.handlePushRegistration(pushCredentials: pushCredentials)
+	}
+	
+	func pushRegistry(_ registry: PKPushRegistry, didReceiveIncomingPushWith payload: PKPushPayload, for type: PKPushType, completion: @escaping () -> Void) {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ClearKeep/ck-ios/blob/fe9f474e9dacdb67166a3b18715e917932515a0f/ClearKeep/App/System/AppDelegate.swift#L122"></span>Function should have complexity 10 or less: currently complexity equals 13<br />
<code>cyclomatic_complexity</code> <code>AppDelegate.swift:122</code></td>
    </tr>
  </tbody>
</table>